File: ios/my_event_emitter.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(MyEventEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)

@end

File: ios/my_event_emitter.swift
import Foundation

@objc(MyEventEmitter)
class MyEventEmitter: RCTEventEmitter {
  
  var hasListener: Bool = false
  
  @objc override func startObserving() {
    hasListener = true
    print("startObserving()");
  }
  
  @objc override func stopObserving() {
    hasListener = false
    print("stopObserving()");
  }
  
  
  @objc override func supportedEvents() -> [String]! {
    return [
      "image-available"
    ];
  }
  
  @objc override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
